app/controllers/episode_player_controller.rb

```diff
@@ -5,10 +5,14 @@ def show
 
     authorize @episode, :show?
 
-    @player_options = {}
-    @player_options[:embed_player_type] = params[:embed_player_type] || "standard"
-    @player_options[:embed_player_theme] = params[:embed_player_theme] || "dark"
-    @player_options[:accent_color] = params[:accent_color] || ["#ff9600"]
-    @player_options[:episode_guid] = Episode.find_by_guid!(params[:episode_id])
+    @player_options = episode_player_params
+  end
+
+  private
+
+  def episode_player_params
+    nilify params
+      .permit(:embed_player_type, :max_width, :embed_player_theme, :accent_color)
+      .with_defaults(EmbedPlayerHelper::DEFAULT_OPTIONS)
```

Comment:
So now, Turbo will GET this action with `?embed_player_type=card`, we regenerate the url/iframe, and re-render the page.

Reply:
Yep - started with all serve side rendering, then added that `data-morph=false` thing to prevent turbo from clobbering the iframe.  And delegate those iframe updates to the stimulus controller.